### PR TITLE
Move the configuration files to versioned target directories

### PR DESCRIPTION
**Pull request by harshula**
* Add more information to the README
* Use symlinking of configurations files instead of $SPACK_USER_CONFIG_PATH
* Use $SPACK_DISABLE_LOCAL_CONFIG to disable ~/spack

### CODE REVIEW

**Review by CodeGat — APPROVED** [review]
Reviewed in person.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,25 @@
...
 
-Shared spack configuration for building software on gadi@NCI
+This repository contains:
+
+* Spack configuration files required by the Spack instance(s) maintained by ACCESS-NRI on Gadi and ACCESS-NRI's [CI Docker image](https://github.com/ACCESS-NRI/build-ci/).
+  * The `ci` directory is to be used by the CI Docker image.
+  * The `gadi` directory is to be used in Gadi deployments.
+  * The `common` directory is a subset of required configuration files and should not be used directly.
+
+* A dot source file, `spack-start.bash`, for developers to setup their `bash` environment to directly run `spack` commands using ACCESS-NRI's Spack instance(s). Execute: `. spack-start.bash`.
```
Comment (aidanheerdegen) [inline README.md:10]:
```suggestion
* A dot source file, `spack-start.bash`, for maintainers to setup their `bash` environment to directly run `spack` commands using ACCESS-NRI's Spack instance(s). Execute: `. spack-start.bash`. Note: this is not for general users.
```

**Review by aidanheerdegen — APPROVED** [review]
Just a suggestion to make it clear `spack-start.bash` isn't for general users.

File: README.md (hunk 1 above)
Comment (harshula) [reply README.md]: This script in its current state is for general users. In the future, for advanced users, we may need to provide a customisable script or create a mechanism to read a configuration file from their home directory. Please see the updated README.

**Review by CodeGat — APPROVED** [review]
Those updates were good. LGTM.

File: README.md (hunk 1 above)
Comment (aidanheerdegen) [reply README.md]: Good point. The updated version is clearer, thanks.

**Pull request merged**